components/style/values/specified/list.rs

```diff
@@ -8,6 +8,44 @@ use cssparser::{Parser, Token};
 use parser::{Parse, ParserContext};
 use std::fmt;
 use style_traits::{ParseError, StyleParseErrorKind, ToCss};
+use values::{Either, None_};
+use values::specified::UrlOrNone;
+
+/// Specified and computed `list-style-image` property.
+///
+#[derive(Clone, Debug, MallocSizeOf, PartialEq, ToCss)]
+pub struct ListStyleImage(pub UrlOrNone);
+
+// FIXME(nox): This is wrong, there are different types for specified
+// and computed URLs in Servo.
+trivial_to_computed_value!(ListStyleImage);
+
+impl ListStyleImage {
+    /// Initial specified value for `list-style-image`.
+    ///
+    #[inline]
+    pub fn get_initial_specified_value() -> ListStyleImage {
+        ListStyleImage(Either::Second(None_))
+    }
+}
+
+impl Parse for ListStyleImage {
+    fn parse<'i, 't>(context: &ParserContext, input: &mut Parser<'i, 't>)
+                         -> Result<ListStyleImage, ParseError<'i>> {
+        #[cfg(feature = "gecko")]
+        {
+            let mut value = input.try(|input| UrlOrNone::parse(context, input))?;
+            if let Either::First(ref mut url) = value {
+                url.build_image_value();
+            }
```

Comment:
let's `return Ok(ListStyleImage(value))` here so it's clearer that the code below doesn't really run, I think, or `#[allow(unused_mut)]`